### ISSUE TITLE
Question: How to bind a class data members to gravity

### ISSUE THREAD

**Issue by untoldengine** [marcobambini/gravity#357]
Hi @marcobambini 

I'm thinking of using Gravity as the scripting language in my C++ game engine. I've been playing around with Gravity and how to use it with my engine. Everything seems to be working fine and I have to say, the C++ binding is a lot nicer than Lua.

So here is my problem, I'm able to bind my C++ classes using the example you provided (thank you, btw). But I can't figure out how to bind my class data members. 

So let's take for example a custom vector class:

```
class Vector3n{
public:
 float x;
 float y;
 float z;

//...constructors

float magnitude(){...}
}
```

I'm able to bind the vector3n class and the magnitude method using 

```
gravity_class_bind(...)
```

In the gravity script, I'm able to do something like this:

```
func update(){
var a=Vector3n(1.0,2.0,1.0);
var m=a.magnitude();
}
```

But now, what if I want to access one of the data members of my vector class in the gravity script

```
func update(){
var a=Vector3n(1.0,2.0,1.0);
var myX=a.x;  //ERROR: I expect to get 1.0
}
```

Which binding function do you have to use to bind class data members?

**Comment by marcobambini** [comment]
Hi @untoldengine,

I updated the C++ with some code that shows you how to expose a C++ property (https://github.com/marcobambini/gravity/commit/e9ce16ce811601501829526256759a16335d8908).
In this example I exposed the length property.

The trick is to use a special closure with a getter and a setter.
If you set the setter to NULL you have a read-only property.

I think it is a powerful way to expose properties but I am open to suggestions and fixes if you think that there could be a better way.

Can't wait to see your C++ game engine.
